src/lib/storage/storage_manager.cpp

```diff
@@ -10,38 +10,33 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
-void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-}
+void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) { _tables[name] = table; }
 
-void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-}
+void StorageManager::drop_table(const std::string& name) { _tables.erase(name); }
```

Comment:
According to the tests this is supposed to throw an exception. But of course with a `map` it doesn't. What to do here guys? @Philipp-Bode @nilsthamm

Reply:
Probably just check against the number of deleted elements and throw manually, right?
Not necessarily super pretty, but OK enough, I'd say.

Reply:
Okay. sec.

Reply:
Done.